python/ray/serve/api.py

```diff
@@ -571,3 +572,100 @@ def delete(name: str, _blocking: bool = True):
     """
     client = get_global_client()
     client.delete_apps([name], blocking=_blocking)
+
+
+@PublicAPI(stability="alpha")
+def multiplexed(
+    func: Optional[Callable[..., Any]] = None, max_num_models_per_replica: int = -1
+):
+    """[EXPERIMENTAL] Defines a function or method used to load multiplexed
+    models in a replica.
+
+    The function can be standalone function or a method of a class. The
+    function must have exactly one argument, the model id of type `str` for the
+    model to be loaded.
+
+    User have to define the function with `async def` and the function must be
+    an async function. It is recommended to define coroutines for long running
+    IO tasks in the function to avoid blocking the event loop.
+
+    The multiplexed function is called to load a model with the given model ID when
+    necessary.
+
+    When the number of models in one replica is larger than max_num_models_per_replica,
+    the models will be unloaded using an LRU policy.
+
+    If you want to release resources after the model is loaded, you can define
+    a `__del__` method in your model class. The `__del__` method will be called when
+    the model is unloaded.
+
+    Example:
+
+    .. code-block:: python
+            from ray import serve
+
+            @serve.deployment
+            class MultiplexedDeployment:
+
+                def __init__(self):
+                    # Define s3 base path to load models
+                    self.s3_base_path = "s3://my_bucket/my_models"
+
+                @serve.multiplexed(max_num_models_per_replica=5)
+                async def load_model(self, model_id: str) -> Any:
+                    # Load model with the given tag
+                    # You can use any model loading library here
+                    # and return the loaded model. load_from_s3 is
+                    # a placeholder function.
+                    return load_from_s3(model_id)
+
+                async def __call__(self, request):
+                    # Get model id from request context
+                    model_id = serve.get_multiplexed_model_id()
+                    # Load model with the given model id
+                    model = await self.load_model(model_id)
+                    # Call the model
+                    return model(request)
```

Comment:
```suggestion
                    return model(request)
```
obvious, too many comments makes it harder to read IMO